src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_RankSumTest.java

```diff
@@ -0,0 +1,121 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import org.apache.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.utils.genotyper.MostLikelyAllele;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+
+import java.util.*;
+
+/**
+ * Allele-specific implementation of rank sum test annotations
+ */
+public abstract class AS_RankSumTest extends RankSumTest implements ReducibleAnnotation {
+    private static final Logger logger = Logger.getLogger(AS_RankSumTest.class);
+    protected static final String splitDelim = "\\|"; //String.split takes a regex, so we need to escape the pipe
+    protected static final String printDelim = "|";
+    protected static final String reducedDelim = ",";
+
+    public List<VCFInfoHeaderLine> getDescriptions() {
+        //TODO only raw for now
+//        if (AnnotationUtils.walkerRequiresRawData(callingWalker))
+            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getRawKeyName()));
+//        else
+//            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getKeyNames().get(0)));
+    }
+
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                         final VariantContext vc,
+                                         final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap) {
+        return annotateRawData(ref, vc, stratifiedPerReadAlleleLikelihoodMap);
+    }
+
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap ) {
+        if ( perReadAlleleLikelihoodMap == null) {
+            return null;
```

Comment:
Previous version returned an empty `Map` in this case -- is the change intentional / documented in the method contract?